upgrades/v5.0.4/README.md

```diff
@@ -0,0 +1,257 @@
+If you use Cosmovisor to upgrade, by default, Cosmovisor will backup your data upon upgrade. See below [upgrade using cosmovisor](#method-ii-upgrade-using-cosmovisor) section.
+
+It is critically important for validator operators to back-up the `.neutrond/data/priv_validator_state.json` file after stopping the neutrond process. This file is updated every block as your validator participates in consensus rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs to be restarted.
+
+### Current runtime
+
+The Neutron mainnet network, `neutron-1`, is currently running [Neutron v5.0.2](https://github.com/neutron-org/neutron/releases/tag/v5.0.2). We anticipate that operators who are running on v5.0.2, will be able to upgrade successfully. Validators are expected to ensure that their systems are up-to-date and capable of performing the upgrade. This includes running the correct binary, or if building from source, building with go `1.22`.
+
+### Target runtime
+
+The Neutron mainnet network, `neutron-1`, will run [Neutron v5.0.4](https://github.com/neutron-org/neutron/releases/tag/v5.0.4). Operators _**MUST**_ use this version post-upgrade to remain connected to the network.
+
+## Upgrade steps
+
+There are 2 major ways to upgrade a node:
+
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://pkg.go.dev/cosmossdk.io/tools/cosmovisor)
+    - Either by manually preparing the new binary
+    - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+## Create the updated Neutron binary of v5.0.4
+
+### Go to neutron directory if present else clone the repository
+
+```shell
+   git clone https://github.com/neutron-org/neutron.git
+```
+
+### Follow these steps if neutron repo already present
+
+```shell
+   cd $HOME/neutron
+   git pull
+   git fetch --tags
+   git checkout v5.0.4
+   make install
+```
+
+### Check the new neutron version, verify the latest commit hash
+```shell
+   $ neutrond version --long
+    commit: 4ae1215c197f5d00f2462f5ce48cf1eaf68bf938
+    cosmos_sdk_version: v0.50.11-neutron
+    go: go version go1.22.9 linux/amd64
+    name: neutron
+    version: 5.0.4
+   ...
+```
+
+### Or check checksum of the binary if you decided to [download it](https://github.com/neutron-org/neutron/releases/tag/v5.0.4)
+
+```shell
+$ shasum -a 256 neutrond-linux-amd64
+40663d2a113d639eedfc29670d0c4996c5143386dabfd559624b4a9679959a03  neutrond-linux-amd64
+```
+
+
+### Make sure you are using the proper version of libwasm
+
+You can check the version you are currently using by running the following command:
+```
+$ neutrond q wasm libwasmvm-version
+
+2.1.3
+```
+The proper version is `2.1.3`.
+
+**If the version on your machine is different you MUST change it immediately!**
```

Comment:
for me it returns 2.1.4:

```
neutrond q wasm libwasmvm-version
2.1.4
```

couldn't suggest changes because of ```. make sure to update the version below in the link and in the snippet, too